torch/_torch_docs.py

```diff
@@ -13897,3 +13897,82 @@ def merge_dicts(*dicts):
 are freshly created instead of aliasing the input.
 """,
 )
+
+
+for unary_base_func_name in (
+    "exp",
+    "sqrt",
+    "abs",
+    "acos",
+    "asin",
+    "atan",
+    "ceil",
+    "cos",
+    "cosh",
+    "erf",
+    "erfc",
+    "expm1",
+    "floor",
+    "log",
+    "log10",
+    "log1p",
+    "log2",
+    "neg",
+    "tan",
+    "tanh",
+    "sin",
+    "sinh",
+    "round",
+    "lgamma",
+    "frac",
+    "reciprocal",
+    "sigmoid",
+    "trunc",
+):
+    unary_foreach_func_name = f"_foreach_{unary_base_func_name}"
+    add_docstr(
+        getattr(torch, unary_foreach_func_name),
+        r"""
+{}(self: List[Tensor]) -> List[Tensor]
+
+Apply :func:`torch.{}` to each Tensor of the input list.
+
+If the input tensors meet a certain condition, for example, they have the same dtype,
+are hosted on the same cuda device, and contiguous (including channels_last and channels_last_3d),
+then the operation done with fewer cuda kernel calls.
+
+.. note::
+    Only CPU and CUDA are supported. Forward-mode AD is not supported.
+
+Args:
+    self (list of Tensors): Input list of Tensors. Each Tensor can have an arbitrary shape, dtype, device, and strides.
+    """.format(
+            unary_foreach_func_name, unary_base_func_name
+        ),
+    )
+    unary_inplace_foreach_func_name = f"{unary_foreach_func_name}_"
+    add_docstr(
+        getattr(torch, unary_inplace_foreach_func_name),
+        r"""
+{}(self: List[Tensor]) -> None
+
+In-place version of :func:`torch.{}`
+
+Args:
+    self (list of Tensors): Input list of Tensors.
+    """.format(
+            unary_inplace_foreach_func_name, unary_foreach_func_name
+        ),
+    )
+
+add_docstr(
+    torch._foreach_zero_,
```

Comment:
This one doesn't need to be special cased? torch.zero_ does exist.

Reply:
as `_foreach_zero` the outplace one doesn't exist.

Reply:
We can special case it above then to only generate these docs if the corresponding `torch._foreach_*` op exists?

Reply:
I don't understand. I handled torch._foreach_zero_ differently because there's no `torch._foreach_zero`. The for-loop above is to generate the docs of `torch._foreach_foo` and `torch._foreach_foo_`.

What would you recommend? Defining `torch._foreach_zero` for the parity with the other unary foreach functions?

Reply:
If the for loop above, you could have a guard for each definition: `if hasattr(torch, "_foreach_foo")` then add the doc for it.
That will remove the need to special case zero (and work for others as well in the future).